File: src/types/price-analysis.ts
export interface PriceAnalysisItem {
  product_id: number
  product_name: string
  product_sku: string
  category: string
  supplier_id: number
  supplier_name: string
  last_supplier_price: number
  current_selling_price: number
  current_cost_price: number
  margin_percent: number
  profit_per_unit: number
  last_purchased: string | null
}

export interface SupplierPriceAnalysis {
  supplier_id: number
  supplier_name: string
  items: PriceAnalysisItem[]
  avg_margin_percent: number
  total_items: number
}

File: backend/app/schemas/price_analysis.py
from pydantic import BaseModel


class PriceAnalysisItem(BaseModel):
    product_id: int
    product_name: str
    product_sku: str
    category: str
    supplier_id: int
    supplier_name: str
    last_supplier_price: float
    current_selling_price: float
    current_cost_price: float
    margin_percent: float
    profit_per_unit: float
    last_purchased: str | None

    model_config = {"from_attributes": True}


class SupplierPriceAnalysisResponse(BaseModel):
    supplier_id: int
    supplier_name: str
    items: list[PriceAnalysisItem]
    avg_margin_percent: float
    total_items: int
